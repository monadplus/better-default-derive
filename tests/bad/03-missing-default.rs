use better_default_derive::Default;

#[derive(Debug, PartialEq, Eq)]
struct Foo(u8);

#[derive(Debug, PartialEq, Eq, Default)]
struct Bar {
    foo: Foo, 
    bar: u8,
}

fn main() {}
