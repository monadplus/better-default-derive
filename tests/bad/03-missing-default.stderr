error[E0599]: no function or associated item named `default` found for struct `Foo` in the current scope
 --> tests/bad/03-missing-default.rs:6:32
  |
4 | struct Foo(u8);
  | ---------- function or associated item `default` not found for this struct
5 |
6 | #[derive(Debug, PartialEq, Eq, Default)]
  |                                ^^^^^^^ function or associated item not found in `Foo`
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following trait defines an item `default`, perhaps you need to implement it:
          candidate #1: `Default`
  = note: this error originates in the derive macro `Default` (in Nightly builds, run with -Z macro-backtrace for more info)
